src/data/finsupp/big_operators.lean

```diff
@@ -0,0 +1,170 @@
+  s.sum.support ⊆ finset.sup s.to_finset finsupp.support :=
+begin
+  induction s using quot.induction_on,
+  simpa using list.support_sum_subset _
+end
+
+lemma finset.support_sum_subset [add_comm_monoid M]
+  (s : finset (ι →₀ M)) :
+  (s.sum id).support ⊆ finset.sup s finsupp.support :=
+by { classical, convert multiset.support_sum_subset s.1; simp }
+
+lemma list.mem_foldr_sup_support_iff [has_zero M] {l : list (ι →₀ M)} {x : ι} :
+  x ∈ (l.map finsupp.support).foldr (⊔) ∅ ↔ ∃ (f : ι →₀ M) (hf : f ∈ l), x ∈ f.support :=
+begin
+  simp only [finset.sup_eq_union, list.foldr_map, finsupp.mem_support_iff, exists_prop],
+  induction l with hd tl IH,
+  { simp },
+  { simp only [IH, list.foldr_cons, finset.mem_union, finsupp.mem_support_iff, list.mem_cons_iff],
+    split,
+    { rintro (h|h),
+      { exact ⟨hd, or.inl rfl, h⟩ },
+      { exact h.imp (λ f hf, hf.imp_left or.inr) } },
+    { rintro ⟨f, rfl|hf, h⟩,
+      { exact or.inl h },
+      { exact or.inr ⟨f, hf, h⟩ } } }
+end
+
+lemma multiset.mem_foldr_sup_support_iff [has_zero M] {s : multiset (ι →₀ M)} {x : ι} :
+  x ∈ (s.map finsupp.support).fold (⊔) ∅ ↔ ∃ (f : ι →₀ M) (hf : f ∈ s), x ∈ f.support :=
+quot.induction_on s $ λ _, list.mem_foldr_sup_support_iff
+
+lemma finset.mem_sup_support_iff [has_zero M] {s : finset (ι →₀ M)} {x : ι} :
+  x ∈ s.sup finsupp.support ↔ ∃ (f : ι →₀ M) (hf : f ∈ s), x ∈ f.support :=
+multiset.mem_foldr_sup_support_iff
+
+lemma list.support_sum_eq [add_monoid M] [decidable_eq (ι →₀ M)] (l : list (ι →₀ M))
+  (hl : (l.map finsupp.support).pairwise disjoint) :
+  l.sum.support = finset.sup l.to_finset finsupp.support :=
+begin
+  induction l with hd tl IH,
+  { simp },
+  { rw [list.map_cons, list.pairwise_cons] at hl,
+    simp only [list.sum_cons, list.to_finset_cons, finset.sup_insert, finset.sup_eq_union],
+    rw [finsupp.support_add_eq, IH hl.right],
+    suffices : disjoint hd.support (tl.to_finset.sup finsupp.support),
+    { exact finset.disjoint_of_subset_right (list.support_sum_subset _) this },
+    { rw finset.disjoint_sup_right,
+      intros f hf,
+      simp only [ne.def, list.mem_to_finset, list.mem_filter] at hf,
+      refine hl.left _ _,
+      simp only [ne.def, list.mem_map, list.mem_filter],
+      exact ⟨f, hf, rfl⟩ } }
+end
+
+lemma multiset.support_sum_eq [add_comm_monoid M] [decidable_eq (ι →₀ M)] (s : multiset (ι →₀ M))
+  (hs : (s.map finsupp.support).pairwise disjoint) :
+  s.sum.support = finset.sup s.to_finset finsupp.support :=
+begin
+  induction s using quot.induction_on,
+  obtain ⟨l, hl, hd⟩ := hs,
+  convert list.support_sum_eq _ _,
+  { simp },
+  { apply_instance },
+  { simp only [multiset.quot_mk_to_coe'', multiset.coe_map, multiset.coe_eq_coe] at hl,
+    exact hl.symm.pairwise hd disjoint.symm }
+end
+
+lemma list.pairwise_of_coe_to_finset_pairwise {α β} [decidable_eq α] {r : β → β → Prop}
+  {f : α → β} {l : list α} (hl : (l.to_finset : set α).pairwise (r on f)) (hn : l.nodup) :
+  l.pairwise (r on f) :=
```

Comment:
These lemmas should be moved to `data/finset/pairwise.`.

Reply:
Moved.